### PR TITLE
sqlserver compression / encryption

### PR DESCRIPTION
**Pull request by mialinx**
### SQLServer

Enables WAL-G standard compression and encryption methods for SQLServer.
Option `WALG_COMPRESSION_METHOD=sqlserver`  has special meaning: WAL-G will request SQLServer to compress data itself.

### CODE REVIEW

File: internal/databases/sqlserver/blob/server.go [hunk 1]
```diff
@@ -72,6 +80,21 @@ func NewServer(folder storage.Folder) (*Server, error) {
 	bs.server = http.Server{Addr: bs.endpoint, Handler: bs}
 	bs.indexes = make(map[string]*Index)
 	bs.leases = make(map[string]Lease)
+	compressor, err := internal.ConfigureCompressor()
+	if err != nil {
+		if _, ok := err.(internal.UnknownCompressionMethodError); !ok {
```
Comment (usernamedt) [inline internal/databases/sqlserver/blob/server.go:85]: Maybe add some warning log to notify the user that WAL-G failed to configure the compressor?

File: internal/databases/sqlserver/blob/server.go [hunk 2]
```diff
@@ -698,3 +742,13 @@ func (bs *Server) AcquireLock() (io.Closer, error) {
 	}
 	return lock, nil
 }
+
+func (bs *Server) validateBlobCompressionEncryption(idx *Index) error {
+	if idx.Encryption != bs.encryption {
+		return fmt.Errorf("blob encryption (%s) does not match configured (%s)", idx.Encryption, bs.encryption)
+	}
+	if idx.Compression != bs.compression {
```
Comment (usernamedt) [inline internal/databases/sqlserver/blob/server.go:750]: Let's assume the following case: Index with some encryption configured but without compression. Server has both encryption and compression enabled. Encryption check is passed, but compressors are obviously mismatched. In this setup, WAL-G will fail to download the uncompressed encrypted file. Maybe allow this scenario (with some warnings)?

Also, maybe go even further and allow all currently supported compressors without the requirement that it should match the currently configured one?

Comment (mialinx) [reply internal/databases/sqlserver/blob/server.go:751]: Changing compression algorithm - one part of the problem. The second one is changing encryption.
The proper solution is to keep metainfo with files (as file extension or in some JSON) and **choose** decompressor/crypter by it's name. But it's not possible for the crypter at the moment.

I wouldn't like to solve only half of the problem, especially it's pretty hypothetical for the SQLServer.

File: internal/databases/sqlserver/blob/server.go (hunk 1 above)
Comment (mialinx) [reply internal/databases/sqlserver/blob/server.go]: Agree.
Will return error unless **special** builtin compression method specified.

**Review by usernamedt — APPROVED** [review]
LGTM

**Review by ostinru — APPROVED** [review]
no criminality

**Pull request merged**